test/test_unary_ufuncs.py

```diff
@@ -33,6 +202,190 @@ def test_unsupported_dtypes(self, device, dtype, op):
         with self.assertRaises(RuntimeError):
             op(t)
 
+    # Tests bool tensor negation raises the correct error
+    def test_neg_error_message(self, device):
+        self.assertRaisesRegex(
+            RuntimeError,
+            r"Negation, the `\-` operator, on a bool tensor is not supported. "
+            r"If you are trying to invert a mask, use the `\~` or `logical_not\(\)` operator instead.",
+            lambda: - torch.tensor([False, True], device=device))
+
+    @dtypes(*floating_types_and(torch.bfloat16, torch.half))
+    @ops((_fn for _fn in unary_ufuncs if _fn.domain != (None, None)))
+    def test_float_domains(self, device, dtype, op):
+        if not op.supports_dtype(dtype, torch.device(device).type):
+            raise unittest.SkipTest('unsupported dtype')
+
+        eps = (1e-1, 1, 2, 10, 20, 50, 100)
+
+        # Adds smaller values except on bfloat16, since these small differences
+        #   are often too small to represent in bfloat16
+        if dtype is not torch.bfloat16:
+            eps = eps + (1e-5, 1e-3)
+
+        if op.domain[0] is not None:
+            low = op.domain[0]
+            for epsilon in eps:
+                t = torch.tensor(low - epsilon, device=device, dtype=dtype)
+                result = op(t)
+                self.assertEqual(result.item(), float('nan'),
+                                 msg="input:{0}".format(t.item()))
+
+        if op.domain[1] is not None:
+            high = op.domain[1]
+            for epsilon in eps:
+                t = torch.tensor(high + epsilon, device=device, dtype=dtype)
+                result = op(t)
+                self.assertEqual(result.item(), float('nan'),
+                                 msg="input:{0}".format(t.item()))
+
+    # Tests that fn == method == inplace == jit on a simple single tensor input
+    @ops(unary_ufuncs)
+    def test_variant_consistency(self, device, dtype, op):
+        def _fn(t):
+            return op(t)
+
+        t = _make_tensor((5, 5), device, dtype, low=op.domain[0], high=op.domain[1])
+        expected = op(t)
+
+        for alt in (op.get_method(), op.get_inplace(), torch.jit.script(_fn)):
+            if alt is None:
+                with self.assertRaises(RuntimeError):
+                    alt(t.clone())
+
+            actual = alt(t.clone())
+            self.assertEqual(actual, expected)
```

Comment:
"just" equal, or bit-for-bit equal?

Reply:
Good catch. For this we should be able to assert equal with rtol/atol=0. In the contiguity tests below, however, we can't get away with that.